Tools/Taxonomy/Common/Options.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using CommandLine;
+
+namespace Taxonomy
+{
+    internal interface IOptions
+    {
+        [Value(0, MetaName = "source file path", HelpText = "Source file path to process.", Required = true)]
+        string SourceFilePath { get; set; }
+
+        [Value(1, MetaName = "target file path", HelpText = "Target file path to save.", Required = true)]
+        string TargetFilePath { get; set; }
+
+        [Value(2, MetaName = "version", HelpText = "Version string in target file.", Required = true)]
+        string Version { get; set; }
+
+        [Value(3, MetaName = "release date utc", HelpText = "ReleaseDateUtc string in target file.", Required = true)]
+        string ReleaseDateUtc { get; set; }
+    }
+
+    [Verb("generate-cwe", false, HelpText = "Generate CWE Sarif file")]
+    internal class CweOptions : IOptions
+    {
+        public string SourceFilePath { get; set; }
+        public string TargetFilePath { get; set; }
+        public string Version { get; set; }
+        public string ReleaseDateUtc { get; set; }
+    }
+
+    [Verb("generate-owasp", false, HelpText = "Generate OWASP Sarif file")]
+    internal class OwaspOptions : IOptions
+    {
+        public string SourceFilePath { get; set; }
+        public string TargetFilePath { get; set; }
+        public string Version { get; set; }
+        public string ReleaseDateUtc { get; set; }
+    }
+
+    [Verb("generate-nistsp80053", false, HelpText = "Generate NIST SP800-63B Sarif file")]
+    internal class NistSP80053Options : IOptions
+    {
+        public string SourceFilePath { get; set; }
+        public string TargetFilePath { get; set; }
+        public string Version { get; set; }
+        public string ReleaseDateUtc { get; set; }
+    }
+
+    [Verb("generate-nistsp80063b", false, HelpText = "Generate NIST SP800-63B Sarif file")]
+    internal class GenerateNistSP80063BOptions
```

Comment:
>Generate [](http://example.com/codeflow?start=19&length=8)

Remove the generate word.
maybe you could merge this with the nist800-53

Reply:
done. I keep them sepearted for now since one is using folder which contains soucre codes nad the other one is using a file.